homeassistant/components/sensor/plex.py

```diff
@@ -48,34 +50,53 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     plex_host = config.get(CONF_HOST)
     plex_port = config.get(CONF_PORT)
     plex_token = config.get(CONF_TOKEN)
-    plex_url = 'http://{}:{}'.format(plex_host, plex_port)
+    plex_ssl = config.get(CONF_SSL)
+    plex_verify_ssl = config.get(CONF_VERIFY_SSL)
```

Comment:
Why is this needed?

Reply:
Why is what needed?

Reply:
If you mean why am I assigning them to variables it's because the variables are used in the main Class as well as before it and having them as variables looks cleaner, in truth we could pass the whole of config into the main class and use the config options directly but I'm not sure it would be very clean looking.

Reply:
Why do we need to be able to disable SSL?

Reply:
Not everyone has it enabled for their local plex server
The issue this is to fix is that the uri was hardcoded as "http://" so no-one with ssl could use the sensor, this patch corrects that oversight.

Reply:
So as far as I know, if Plex connects locally, it connects over IP and thus never needs SSL. For remote access, they set up a dyndns and add SSL on top of it.

Reply:
See issue: https://github.com/home-assistant/home-assistant/issues/10509

Reply:
Okay, so that's SSL in the cloud. So I understand the SSL config var. I don't understand why we would ever allow disabling SSL verification?

Reply:
99% of the time for Self-Signed Certs or proxy issues.
1% Troubleshooting
 It is NOT recommended and actively discouraged but better to have the option and not need it than need it and not be able to do it.

Reply:
We generally take a lot of care to limit the number of configuration options exposed to the user just to avoid confusion. Let's not preemptively add an option for something that's actively discouraged and breaks security. If they are using self-signed certs they need to properly configure their system to accept them.

Reply:
@armills  

This option already is exposed in the plex media_player component, would it not be preferable to keep parity with the existing implementation within the same product?

Additionally this option is defaulted to True so most people will take the default option here, you would have to explicitly say "No don't check", which to me means you know what you are doing and why.

I dont use the SSL option for this anyway so i'm not particularly fussed, though I do think that where there is a chance someone will need an option we should expose it with a sensible default set. This why people can ignore it if its not applicable but equally if they need it it is there.